pkg/operator/observe_config.go

```diff
@@ -86,6 +88,45 @@ func (c ConfigObserver) sync() error {
 		unstructured.SetNestedStringSlice(observedConfig, etcdURLs, "storageConfig", "urls")
 	}
 
+	// observe configuration from cluster-config-v1
+
+	// get cluster config
+	clusterConfig, err := c.kubeClient.CoreV1().ConfigMaps("kube-system").Get("cluster-config-v1", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error retieving cluster config: %s", err)
+	}
+
+	// get install-config from cluster config
+	installConfigYaml, ok := clusterConfig.Data["install-config"]
+	if !ok {
+		return fmt.Errorf("cluster-config-v1 missing install-config")
+	}
+	installConfig := installer.InstallConfig{}
+	if err := yaml.Unmarshal([]byte(installConfigYaml), &installConfig); err != nil {
+		return fmt.Errorf("error parsing install-config: %s", err)
+	}
+
+	// extract needed values
+	//  data:
+	//   install-config:
+	//     networking:
+	//       podCIDR: 10.2.0.0/16
+	//       serviceCIDR: 10.3.0.0/16
+	restrictedCIDRs := []string{
+		installConfig.Networking.PodCIDR.String(),
+		installConfig.Networking.ServiceCIDR.String(),
+	}
+
+	// set observed values
+	//  admissionPluginConfig:
```

Comment:
wow.  One the one hand I'm glad I now know.  On the other hand, that's going to be a bitch to maintain.

@squeed @knobunc does anyone use this admission plugin to control ranges other than these two CIDRs?  Should they be able to?

Reply:
Not that I know of... but who knows what people have configured.